diff --git a/src/vs/code/electron-main/app.ts b/src/vs/code/electron-main/app.ts
index 43aba01..23b55ea 100644
--- a/src/vs/code/electron-main/app.ts
+++ b/src/vs/code/electron-main/app.ts
@@ -553,6 +553,13 @@ export class CodeApplication extends Disposable {
 	async startup(): Promise<void> {
 		this.logService.debug('Starting !!APP_NAME!!');
 		this.logService.debug(`from: ${this.environmentMainService.appRoot}`);
+
+		this.logService.info('Starting !!APP_NAME!! AI Daemon...');
+		try {
+			require('child_process').spawn(require('path').join(this.environmentMainService.appRoot, '..', '!!BINARY_NAME!!-daemon'), [], { detached: true, stdio: 'ignore' }).unref();
+		} catch (err) {
+			this.logService.error('Failed to start !!APP_NAME!! AI Daemon', err);
+		}
 
 		this.appInstantiationService.invokeFunction(accessor => {
